src/spec_tools/core.cljc

```diff
@@ -101,6 +102,51 @@
         `(map->Type (merge ~info {:hint ~hint :form '~(or (->> pred resolve impl/->sym) pred), :pred ~pred}))))))
 
 ;;
+;; Map Spec
+;;
+
+(defrecord OptionalKey [k])
+(defrecord RequiredKey [k])
+
+(defn opt [k] (->OptionalKey k))
+(defn req [k] (->RequiredKey k))
+
+#?(:clj
+   (defn -map [n m]
+     (reduce-kv
+       (fn [acc k v]
+         (let [required? (fn [x]
```

Comment:
this is utter crap. Wanted to resolve the symbol to be `spec-tools.core/opt` or `spec-tools.core/req`. With CLJS, symbol resolution `(resolve x)` gives `nil` from tests (symbol was `st/opt`). Works on CLJ-side ok.